.github/workflows/labels.yml

```
@@ -31,57 +35,107 @@ jobs:
+                status: 'success',
+                exclude_pull_requests: true
+              })).data.workflow_runs[0]?.created_at
+            })())
+            core.info('cutoff timestamp: ' + cutoff.toISOString())
+
+            // To simplify this action's logic we fetch the pull_request data again below, even if
+            // we are already in a pull_request event's context and would have the data readily
+            // available. We do this by filtering the list of pull requests with head and base
+            // branch - there can only be a single open Pull Request for any such combination.
+            const prEventCondition = !context.payload.pull_request ? undefined : {
+              // "label" is in the format of `user:branch` or `org:branch`
+              head: context.payload.pull_request.head.label,
+              base: context.payload.pull_request.base.ref
             }
 
-            await Promise.all(
-              pull_requests.map(async (pull_request) => {
+            await github.paginate(
+              github.rest.pulls.list,
+              {
+                ...context.repo,
+                state: 'open',
+                sort: 'updated',
+                direction: 'desc',
+                ...prEventCondition
+              },
+              async (response, done) => await Promise.all(response.data.map(async (pull_request) => {
+                const log = (k,v) => core.info(`PR #${pull_request.number} - ${k}: ${v}`)
+
+                log('Last updated at', pull_request.updated_at)
+                if (new Date(pull_request.updated_at) < cutoff) return done()
+
+                const run_id = (await github.rest.actions.listWorkflowRuns({
+                  ...context.repo,
+                  workflow_id: 'eval.yml',
+                  event: 'pull_request_target',
+                  // For PR events, the workflow run is still in progress with literally this job itself.
+                  status: prEventCondition ? 'in_progress' : 'success',
+                  exclude_pull_requests: true,
+                  head_sha: pull_request.head.sha
+                })).data.workflow_runs[0]?.id
+
+                // Newer PRs might not have run Eval to completion, yet. We will catch them on the
+                // next run, no need to wait here.
+                log('Last eval run', run_id)
+                if (!run_id) return;
+
+                const artifact = (await github.rest.actions.listWorkflowRunArtifacts({
+                  ...context.repo,
+                  run_id,
+                  name: 'comparison'
+                })).data.artifacts[0]
+
+                // Instead of checking the boolean artifact.expired, we will give us a minute to
+                // actually download the artifact in the next step and avoid that race condition.
+                log('Artifact expires at', artifact.expires_at)
+                if (new Date(artifact.expires_at) < new Date().getTime() + 60 * 1000) return;
```

Isn't this a type error?

```suggestion
                if (new Date(artifact.expires_at) < new Date(new Date().getTime() + 60 * 1000)) return;
```